app/keeper/internal/net/contribute.go

```diff
@@ -61,6 +61,9 @@ func Contribute(source *workloadapi.X509Source) {
 				Shard:    base64.StdEncoding.EncodeToString(contribution),
 			},
 		)
+		if err != nil {
+			panic("marshalling shard contribution request: " + err.Error())
```

Comment:
Some errors in this function lead to a Fatal and some lead to a Panic. Let me know if this should be a Fatal instead. I have a feeling we ought to refactor this to bubble the error up at somepoint regardless.

Reply:
the current pattern is use log.FatalF or FatalLn instead of panicking.

That might be a remnant of a former proof-of-concept.

I have an action item to go through the entire codebase (for consistency); but haven't gotten there yet :).

Reply:
I will fix this today so we can merge.